### PR TITLE
Add infrastructure overview doc

### PR DESCRIPTION
**Pull request by fgimenez**
Also updates the main readme.

/cc @enp0s3 @rmohr @dhiller 

Signed-off-by: Federico Gimenez <fgimenez@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,86 +1,61 @@
...
 
-   * github/ci/prow-workloads
+* `limiter`: Tool used to control connections for GCE buckets to the outside world
+depending on billing alerts. See [README](limiter/README.md)
 
-   Ansible code for bootstraping prow-workloads cluster with Kubespray
+* `releng/release-tool`: Tool for creating KubeVirt releases
 
-   * github/ci/services
+* `robots`: Several Go based automation tools
```
Comment (dankenigsberg) [inline README.md:35]: Is the language in which they are written is important to be mentioned here?

File: docs/infrastructure-components.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+of how Prow runs jobs take a look at ["Life of a Prow Job"]. To see common Prow
+usage and interactions flow, see the pull request interactions [sequence diagram].
+
+## Layout
+
+![infra-layout](infra-layout.png)
+
+## Clusters
+
+Our infrastructure currently uses 3 main clusters. For each of them we will
```
Comment (dankenigsberg) [inline docs/infrastructure-components.md:18]: Do we have non-main clusters? Shouldn't we list them all here?

How about
```suggestion
Our infrastructure includes the following clusters. For each of them, this document provides the following fields
```

File: docs/infrastructure-components.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+include the following fields:
+
+* Description: brief explanation of the cluster purpose.
+* Context: context name of the cluster in the master kubeconfig used by Prow to
```
Comment (dankenigsberg) [inline docs/infrastructure-components.md:22]:
```suggestion
* Prow context: context name of the cluster in the master kubeconfig used by Prow to
```
maybe?

Comment (dankenigsberg) [reply docs/infrastructure-components.md:22]: can you add a url to the master kubeconfig?

File: docs/infrastructure-components.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+* External services: functionality not running in our infra that is required for
+the normal operation of the workloads running in the cluster.
+* Connected clusters: for each cluster, which others are required for its normal
+operation.
+* Exposed services: functionality accessible from outside the cluster, usually as
+web applications.
+
+### Control plane
+
+#### Description
```
Comment (dankenigsberg) [inline docs/infrastructure-components.md:36]: to me, the "Description" section is pointless. I'd just have the text follow the name of the cluster

File: docs/infrastructure-components.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+#### Exposed services
+
+None.
+
+### phx-prow
+
+#### Description
+
+Old cluster where Prow control plane run initially. It is still used to run some
```
Comment (dankenigsberg) [inline docs/infrastructure-components.md:133]:
```suggestion
Old cluster where Prow control plane used to run. It is still used to run some
```
better English

File: docs/infrastructure-components.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+external registries.
+
+#### External services
+
+* Quay: we mainly use quay as our container image registry, the images used for
+testing and the image artifacts are stored here.
+
+#### Connected clusters
+
+Control plane, it sends prow jobs here to be executed and retrieves its state.
```
Comment (dankenigsberg) [inline docs/infrastructure-components.md:152]:
```suggestion
The Control plane cluster sends prow jobs here to be executed and retrieves its state.
```

File: docs/infrastructure-components.md [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+CI/CD systems and related services, like the monitoring stack or applications for
+indexing and querying build logs.
+
+The CI/CD system is based on [Prow], kubernetes CI system. For a brief overview
+of how Prow runs jobs take a look at ["Life of a Prow Job"]. To see common Prow
+usage and interactions flow, see the pull request interactions [sequence diagram].
+
+## Layout
+
+![infra-layout](infra-layout.png)
```
Comment (dankenigsberg) [inline docs/infrastructure-components.md:14]: can we have this in an editable SVG?

**Review by dankenigsberg — COMMENTED** [review]
(very partial review)

File: docs/infrastructure-components.md (hunk 2 above)
Comment (fgimenez) [reply docs/infrastructure-components.md]: Sounds good :+1:

File: docs/infrastructure-components.md (hunk 3 above)
Comment (fgimenez) [reply docs/infrastructure-components.md]: Sure done. It is part of the automation secrets, though. There's  a good explanation about how it is handled in the document about adding new clusters proposed here #1487, added a link to it too.

File: docs/infrastructure-components.md (hunk 4 above)
Comment (fgimenez) [reply docs/infrastructure-components.md]: Sure done

File: README.md (hunk 1 above)
Comment (fgimenez) [reply README.md]: Not at all, removed.

**Comment by fgimenez** [comment]
@dankenigsberg all the suggestions have been addressed, PTAL.

File: docs/infrastructure-components.md [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+usage and interactions flow, see the pull request interactions [sequence diagram].
+
+## Layout
+
+![infra-layout](infra-layout.svg)
+
+## Clusters
+
+Our infrastructure includes the following clusters. For each of them, this
+document provides the following fields include the following fields:
```
Comment (enp0s3) [inline docs/infrastructure-components.md:19]: @fgimenez I think `this
document provides the following fields include the following fields` has duplication. Did you mean `this document provides the following fields`?

**Review by enp0s3 — CHANGES_REQUESTED** [review]
Thank you! besides a small correction everything looks great! please see a comment below

File: docs/infrastructure-components.md (hunk 8 above)
Comment (fgimenez) [reply docs/infrastructure-components.md]: Indeed good catch! Should be fixed now, PTAL.

**Comment by dankenigsberg** [comment]
I know it's considered a bit silly and old fashion, but I care about future `git log`. Would you squash it to something more readable?

As for the text, I'm happy with it now.

**Comment by fgimenez** [comment]
> I know it's considered a bit silly and old fashion, but I care about future `git log`. Would you squash it to something more readable?

Absolutely, done.

**Review by enp0s3 — APPROVED** [review]
Thank you!
/lgtm
/approve

**Pull request merged**